public/app/features/panel/panellinks/link_srv.ts

```diff
@@ -202,3 +205,24 @@ export function getLinkSrv(): LinkService {
 }
 
 coreModule.service('linkSrv', LinkSrv);
+
+/**
+ * Helper to get all links from datasoure already interpolated.
+ * @param datasource
+ * @param rowData Some data that will be passed to the datasource to extract additional scoped vars. See
+ *  getDerivedFields
+ */
+export async function getLinksFromDataSource(
```

Comment:
Utils to get the links from datasource. This is done lazily as in LogsPanel we show it only when details view is opened.